v2/samples/apimanagement/v1api20220801/v1api20220801_authorizationprovider.yaml

```
@@ -0,0 +1,20 @@
+apiVersion: apimanagement.azure.com/v1api20220801
+kind: AuthorizationProvider
+metadata:
+  name: sampleapimauthorizationprovider
+  namespace: default
+spec:
+  displayName: "sampleauthcode"
+  identityProvider: "aad"
+  oauth2:
+    grantTypes:
+      authorizationCode:
+        clientIdFromConfig:
+          name: identity-settings
+          key: clientId
+        clientSecret: # This is the name/key of a Kubernetes secret in the same namespace
```

Did we come up with this name? If so, we may want to rename this to clarify this comes from Kubernetes secret.

For example:
```suggestion
        clientSecretKeyRef: # This is the name/key of a Kubernetes secret in the same namespace
```

This aligns with `secretKeyRef` how secrets are pulled to an environment variable for a Kubernetes deployment. Unless this is the pattern in ASO overall for interacting with Kubernetes secret of course

---

This will be addressed by #3435, I need to change it when it is supported